lifelines/fitters/__init__.py

```diff
@@ -422,11 +420,9 @@ def _negative_log_likelihood_interval_censoring(self, params, Ts, E, entry, weig
             ll
             + np.clip(
                 censored_weights
-                * anp.log(
-                    self._survival_function(params, censored_starts) - self._survival_function(params, censored_stops)
-                ),
-                1e-50,
-                1,
+                * anp.log(self._survival_function(params, censored_starts) - self._survival_function(params, censored_stops)),
+                -1e50,
+                1e50,
```

Comment:
this was the bug. I must have not seen the `log` transformation happening, and assumed this was clipping a difference of probabilities.

Reply:
Reverting for now, will attempt again later.

Reply:
I've added a test as well.